mesonbuild/build.py

```diff
@@ -1120,10 +1050,10 @@ def process_kwargs(self, kwargs):
         self.add_deps(deplist)
         # If an item in this list is False, the output corresponding to
         # the list index of that item will not be installed
-        self.install_dir = typeslistify(kwargs.get('install_dir', []),
+        self.install_dir = typeslistify(kwargs.get('install_dir') or [],
                                         (str, bool))
```

Comment:
IIRC (but need to double check) install_dir can be `False` for CustomTarget but not for BuildTarget, is that right? So I think we can change to `stringlistify()` here and make sure the typing says it's a `T.List[str]`.

Reply:
Oh, it can be for BuildTarget because vala targets generate multipel outputs. I'm planning to get rid of that eventually, but it's invasive

Reply:
I'm that case you can't use `or` here

Reply:
I guess this change isn't actually needed, since what it was added for was the original pass before i discovered the madness of vala targets, so I'll revert this back to the origin `get()` call with no `or`. I don't think the `or` is actually hurting anything, but the change is unnecessary.

I've reverted the change and added a TODO explaining why the code is as it is, and that we should replace it with explicit keyword arguments.